src/key.rs

```diff
@@ -0,0 +1,137 @@
+use crossterm::event::{KeyCode, KeyModifiers};
+use serde::{de, Deserialize};
+
+#[derive(Debug, Deserialize, Default)]
+#[serde(default)]
+pub struct KeyMap {
+    #[serde(deserialize_with = "deseralize_key")]
+    pub remove_previous_word: Option<Key>,
+    #[serde(deserialize_with = "deseralize_key")]
+    pub remove_previous_char: Option<Key>,
+    #[serde(deserialize_with = "deseralize_key")]
+    pub next_word: Option<Key>,
+}
+
+fn deseralize_key<'de, D>(deserializer: D) -> Result<Option<Key>, D::Error>
+where
+    D: de::Deserializer<'de>,
+{
+    struct KeyVisitor;
+    impl<'de> de::Visitor<'de> for KeyVisitor {
+        type Value = Option<Key>;
+
+        fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+            formatter.write_str("")
+        }
+
+        fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+        where
+            E: de::Error,
+        {
+            Ok(get_key_from_string(v))
+        }
+    }
+
+    return deserializer.deserialize_str(KeyVisitor);
+}
+
+#[derive(Debug)]
+pub struct Key {
+    pub code: KeyCode,
+    pub modifier: KeyModifiers,
+}
+
+impl Default for Key {
+    fn default() -> Self {
+        Self {
+            code: KeyCode::Null,
+            modifier: KeyModifiers::NONE,
+        }
+    }
+}
+
+fn get_key_code_from_string(string: &str) -> KeyCode {
+    if string.len() == 1 {
```

Comment:
This is subtly wrong. `str::len` returns the length in _bytes_, not unicode code points like `char`. You have to use the `str::chars` iterator for this.